src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1064,6 +1066,30 @@ static int client_send_request(sd_dhcp_client *client) {
                         return r;
         }
 
+       if (client->user_class) {
+               r = dhcp_option_append(&request->dhcp, optlen, &optoffset, 0,
+                                       SD_DHCP_OPTION_USER_CLASS,
+                                       strv_length(client->user_class),
+                                       client->user_class);
+                if (r < 0)
+                        return r;
+        }
+
+        ORDERED_HASHMAP_FOREACH(j, client->extra_options, i) {
+                r = dhcp_option_append(&request->dhcp, optlen, &optoffset, 0,
+                                       j->option, j->length, j->data);
+                if (r < 0)
+                        return r;
+        }
+
+        if (!ordered_hashmap_isempty(client->vendor_options)) {
+                r = dhcp_option_append(
+                                &request->dhcp, optlen, &optoffset, 0,
+                                SD_DHCP_OPTION_VENDOR_SPECIFIC,
+                                ordered_hashmap_size(client->vendor_options), client->vendor_options);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
hmm, the very same lines already exist in client_send_discover(). Maybe add some common shared helper function client_append_common_discover_request_options()? i.e. a helper that appens all options included in both request + discover?

Reply:
Ok .